lvgl_tft/st7789.c

```diff
@@ -85,52 +83,68 @@ void st7789_init(void)
         {0, {0}, 0xff},
     };
 
-    //Initialize non-SPI GPIOs
-    gpio_pad_select_gpio(ST7789_DC);
-    gpio_set_direction(ST7789_DC, GPIO_MODE_OUTPUT);
-
-#if !defined(ST7789_SOFT_RST)
-    gpio_pad_select_gpio(ST7789_RST);
-    gpio_set_direction(ST7789_RST, GPIO_MODE_OUTPUT);
-#endif
-
-    //Reset the display
-#if !defined(ST7789_SOFT_RST)
-    gpio_set_level(ST7789_RST, 0);
-    vTaskDelay(100 / portTICK_RATE_MS);
-    gpio_set_level(ST7789_RST, 1);
-    vTaskDelay(100 / portTICK_RATE_MS);
-#else
-    st7789_send_cmd(ST7789_SWRESET);
-#endif
-
-    printf("ST7789 initialization.\n");
+    st7789_reset(drv);
 
     //Send all the commands
     uint16_t cmd = 0;
     while (st7789_init_cmds[cmd].databytes!=0xff) {
-        st7789_send_cmd(st7789_init_cmds[cmd].cmd);
-        st7789_send_data(st7789_init_cmds[cmd].data, st7789_init_cmds[cmd].databytes&0x1F);
+        st7789_send_cmd(drv, st7789_init_cmds[cmd].cmd);
+        st7789_send_data(drv, st7789_init_cmds[cmd].data, st7789_init_cmds[cmd].databytes&0x1F);
         if (st7789_init_cmds[cmd].databytes & 0x80) {
-                vTaskDelay(100 / portTICK_RATE_MS);
+            display_hal_delay(drv, 100);
         }
         cmd++;
     }
 
-    st7789_set_orientation(CONFIG_LV_DISPLAY_ORIENTATION);
+    st7789_enable_backlight(drv, true);
+
+    /* FIXME We're setting up the initial orientation in the cmd array */
+    st7789_set_orientation(drv, ST7789_INITIAL_ORIENTATION);
+}
+
+void st7789_enable_backlight(lv_disp_drv_t *drv, bool backlight)
+{
+#if ST7789_ENABLE_BACKLIGHT_CONTROL
+    uint32_t tmp = 0;
+
+#if (ST7789_BCKL_ACTIVE_LVL==1)
+    tmp = backlight ? 1 : 0;
+#else
+    tmp = backlight ? 0 : 1;
+#endif
+
+    display_hal_backlight(drv, tmp);
+#endif
 }
 
 /* The ST7789 display controller can drive 320*240 displays, when using a 240*240
  * display there's a gap of 80px, we need to edit the coordinates to take into
  * account that gap, this is not necessary in all orientations. */
 void st7789_flush(lv_disp_drv_t * drv, const lv_area_t * area, lv_color_t * color_map)
 {
+    static lv_disp_rot_t prev_rotation = LV_DISP_ROT_NONE;
+
+    lv_disp_t *display = (lv_disp_t *) drv;
+    lv_disp_rot_t rotation = lv_disp_get_rotation(display);
+
+    if (rotation != prev_rotation) {
+        /* Update rotation */
+
+        /* TODO Handle all other rotations */
+        if (LV_DISP_ROT_180 == rotation) {
+            st7789_set_orientation(drv, 1);
+        }
+
+        prev_rotation = rotation;
+    }
```

Comment:
I suggest leaving the rotation feature to another PR

Reply:
Agree, how should I remove this commit, a revert?

Reply:
Fixed with 38bca35